include/deal.II/particles/particle.h

```diff
@@ -0,0 +1,306 @@
+     * ID at the specified location. Note that deal.II
+     * does not check for duplicate particle IDs so the user must
+     * make sure the IDs are unique over all processes.
+     *
+     * @param new_location Initial location of particle.
+     * @param new_reference_location Initial location of the particle
+     * in the coordinate system of the reference cell.
+     * @param new_id Globally unique ID number of particle.
+     */
+    Particle (const Point<dim> &new_location,
+              const Point<dim> &new_reference_location,
+              const types::particle_index new_id);
+
+    /**
+     * Copy-Constructor for Particle, creates a particle with exactly the
+     * state of the input argument. Note that since each particle has a
+     * handle for a certain piece of the property memory, and is responsible
+     * for registering and freeing this memory in the property pool this
+     * constructor registers a new chunk, and copies the properties.
+     */
+    Particle (const Particle<dim> &particle);
+
+    /**
+     * Constructor for Particle, creates a particle from a data vector.
+     * This constructor is usually called after sending a particle to a
+     * different process.
+     *
+     * @param[in,out] begin_data A pointer to a memory location from which
+     * to read the information that completely describes a particle. This
+     * class then de-serializes its data from this memory location and
+     * advance the pointer accordingly.
+     *
+     * @param[in,out] new_property_pool A property pool that is used to
+     * allocate the property data used by this particle.
+     */
+    Particle (const void *&begin_data,
+              PropertyPool &new_property_pool);
+
+    /**
+     * Move constructor for Particle, creates a particle from an existing
+     * one by stealing its state.
+     */
+    Particle (Particle<dim> &&particle);
+
+    /**
+     * Copy assignment operator.
+     */
+    Particle<dim> &operator=(const Particle<dim> &particle);
+
+    /**
+     * Move assignment operator.
+     */
+    Particle<dim> &operator=(Particle<dim> &&particle);
+
+    /**
+     * Destructor. Releases the property handle if it is valid, and
+     * therefore frees that memory space for other particles. (Note:
+     * the memory is managed by the property_pool, and the memory is not
+     * deallocated by this function, it is kept in reserve for other
+     * particles).
+     */
+    ~Particle ();
+
+    /**
+     * Write particle data into a data array. The array is expected
+     * to be large enough to take the data, and the void pointer should
+     * point to the first element in which the data should be written. This
+     * function is meant for serializing all particle properties and
+     * afterwards de-serializing the properties by calling the appropriate
+     * constructor Particle(void *&data, const unsigned int data_size);
+     *
+     * @param [in,out] data The memory location to write particle data
```

Comment:
after doing a search, i realised that the `in/out` question is not really settled. There is already a number of places in documentation which use `in/out`. Sorry for asking you to remove them.
Feel free to use it everywhere you see fit.